scalyr_agent/monitor_utils/k8s.py

```diff
@@ -424,11 +424,11 @@ def __init__(
         for k, v in six.iteritems(labels):
```

Comment:
This is not something to do with this PR, but just thinking out loud - if we call this ``PodInfo`` code relatively often and on many pods it may actually be beneficial to optimize this hash calculation approach a bit.

Reply:
So after checking the code again, I'm not even sure it always works correctly / as expected.

We iterate over and dictionaries and lists before first sorting them. For lists that's probably fine, but for dicts it's not since sorting and as such also iteration order is not guaranteed in Python (and we are not using ``collections.OrderedDict`` either).

Which means that when retrieving and deserializating pod information from Kubernetes API, this digest value may indicate that the pod information has changed, but it actually hasn't and the change is due to the different order of the keys in the dictionary on iteration which of course results in a different hash.

So this could cause unnecessary churn and updating the log configs for the "affected" pods.

Reply:
#733